kale/embed/mpca.py

```diff
@@ -144,52 +144,43 @@ def _fit(self, x):
         proj_mats = []
 
         for i in range(1, n_dims):
-            for j in range(n_samples):
-                # unfold the j_th tensor along the i_th mode
-                x_ji = unfold(x[j, :], mode=i - 1)
-                if i not in phi:
-                    phi[i] = 0
-                phi[i] = phi[i] + np.dot(x_ji, x_ji.T)
+            phi[i] = unfold(x, mode=i)
```

Comment:
Consider a new variable name. This is not longer the same as in the Matlab code or paper so a more descriptive name would be better, e.g. `mode_data_mat`. Moreover, why creating these for all modes and then do the SVD (increasing the memory requirement nx? This for loop should be combined with the for loop below to be more space (and time) efficient.